ios-base/Extensions/ViewExtension.swift

```diff
@@ -53,6 +53,61 @@ extension UIView {
       self?.layoutIfNeeded()
     })
   }
+  
+  // MARK: Constrains Helper
+  
+  func attachHorizontally(
+    subview: UIView,
+    leadingMargin: CGFloat = UI.Defaults.margin,
+    trailingMargin: CGFloat = UI.Defaults.margin
+  ) {
+    NSLayoutConstraint.activate([
+      subview.leadingAnchor.constraint(equalTo: leadingAnchor, constant: leadingMargin),
+      subview.trailingAnchor.constraint(
+        equalTo: trailingAnchor,
+        constant: -trailingMargin
+      )
+    ])
+  }
+  
+  func addSubviews(subviews: [UIView]) {
+    for subview in subviews {
+      addSubview(subview)
+    }
+  }
+  
+  func attachVertically(
+    subview: UIView,
+    topMargin: CGFloat = UI.Defaults.margin,
+    bottomMargin: CGFloat = UI.Defaults.margin
+  ) {
+    NSLayoutConstraint.activate([
+      subview.topAnchor.constraint(equalTo: topAnchor, constant: topMargin),
+      subview.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -bottomMargin)
+    ])
+  }
+  
+  func center(
+    subview: UIView,
+    verticalOffset: CGFloat = 0,
+    horizontalOffset: CGFloat = 0,
+    vertically: Bool = true,
+    horizontally: Bool = true
+  ) {
+    if vertically {
+      subview.centerYAnchor.constraint(
+        equalTo: centerYAnchor,
+        constant: verticalOffset
+      ).isActive = true
+    }
+    
+    if horizontally {
+      subview.centerXAnchor.constraint(
+        equalTo: centerXAnchor,
+        constant: horizontalOffset
+      ).isActive = true
+    }
```

Comment:
this could crash if the view is not added to the hierarchy, which is an assumption this method is doing. Please add a check for that case or add the proper docstrings for the function.

Reply:
I think a better approach would be making it the other way around. Like:
```
func centerInView(parent: UIView, ...) {
  if parent != superview {
    parent.addSubview(self)
  } 
  ...
}
```

This way you also avoid the issues with hierarchy.

Reply:
@kstoletniy please see my method's signature. To this function you pass the parent view where you want to center the current instance. It makes more sense since you are calling this function on that instance, not the parent.
i.e `subview.centerInView(parent, ...)`

The point of checking for the parent and adding self as a subview is that you don't trust the caller to read the docs and expect them to crash, also you don't assume they will have the correct view hierarchy.

On a side note, having control paramenters (`vertically`) is a code smell. This can be split in two functions like:
`centerVertically(inParent parent: UIView, ...)` and `centerHorizontally(inParent parent: UIView, ...)`.

Reply:
I think it goes better as is with the rest of the auto layout helpers:

```
  superView.attachHorizontally(subview)
  superView.activateMarginConstraints(n: subview)
```

also I don't see how having control parameters is a code smell? 
but anyways we can update so we have
```
centerVertically()
centerHorizontally()

func center(subview: UIView) {
  centerVertically()
  centerHorizontally()
}

Reply:
IMO the latter reads much better, and the call it is directly related to the current view instance which makes clear the result.
You can read more about control parameters here: https://martinfowler.com/bliki/FlagArgument.html

Reply:
I'm agree with @glm4. It's more readable in that way so I've done a refactor for this methods and attach methods as well. However I didn't add the `addSubview` statement. If the method is called with a wrong view then we add it and center it would end on a misplaced behavior for the caller. Please check the new implementation and let me know your thoughts.